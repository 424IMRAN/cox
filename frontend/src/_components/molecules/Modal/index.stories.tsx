import { Story, Meta } from '@storybook/react'
import { ThemeProvider, makeStyles } from '@material-ui/core/styles'
import { Button, Typography } from '@material-ui/core'
import Modal from '.'
import baseTheme from '../../../core-utils/theme'
import TextField from '../../atoms/TextField'

export default {
  title: 'molecules/Modal',
  component: Modal,
  argTypes: {},
} as Meta

const useStyle = makeStyles({
  root: {
    width: '560px',
  },
})

const Template: Story = ({ ...args }) => {
  const classes = useStyle()
  return (
    <ThemeProvider theme={baseTheme}>
      <Modal handleModalClose={() => {}} open title="test" {...args}>
        <TextField label="Filter Name" fullWidth customStyle={classes.root} />
      </Modal>
    </ThemeProvider>
  )
}

export const BasicModal = Template.bind({})
BasicModal.args = {
  title: 'Add/Rename Filter',
  open: true,
  primaryButton: (
    <Button variant="contained" disabled>
      Primary
    </Button>
  ),
  secondaryButton: <Button variant="outlined">Secondary</Button>,
}

export const PrimaryModal = Template.bind({})
PrimaryModal.args = {
  title: 'Add/Rename Filter',
  open: true,
  primaryButton: (
    <Button variant="contained" disabled>
      Primary
    </Button>
  ),
}
export const NoButtonModal = Template.bind({})
NoButtonModal.args = {
  title: 'Add/Rename Filter',
  open: true,
}
