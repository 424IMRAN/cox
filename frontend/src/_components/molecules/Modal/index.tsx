import React from 'react'
import { Theme, makeStyles } from '@material-ui/core/styles'
import Dialog from '@material-ui/core/Dialog'
import MuiDialogTitle from '@material-ui/core/DialogTitle'
import DialogContent from '@material-ui/core/DialogContent'
import DialogActions from '@material-ui/core/DialogActions'
import IconButton from '@material-ui/core/IconButton'
import CloseIcon from '@material-ui/icons/Close'
import Divider from '@material-ui/core/Divider'
import PropTypes from 'prop-types'
import Typography from '../../atoms/Typography'

const useStyles = makeStyles((theme: Theme) => ({
  root: {
    margin: 0,
  },
  closeButton: {
    position: 'absolute',
    right: theme.spacing(1),
    top: theme.spacing(1),
    color: theme.palette.grey[500],
  },
}))

export interface DialogTitleProps {
  children: string
  onClose: () => void
  buttonStyle: string
}

const DialogTitle = (props: DialogTitleProps) => {
  const { children, onClose, buttonStyle, ...other } = props
  return (
    <MuiDialogTitle {...other}>
      <Typography variant="heading2">{children}</Typography>
      {/* {children} */}
      <IconButton aria-label="close" className={buttonStyle} onClick={onClose}>
        <CloseIcon />
      </IconButton>
    </MuiDialogTitle>
  )
}

type ModalProps = {
  handleModalClose: (close: boolean) => void
  open: boolean
  title: string
  children: React.ReactNode
  primaryButton?: React.ReactNode
  secondaryButton?: React.ReactNode
}

const Modal = ({
  handleModalClose,
  open,
  title,
  children,
  primaryButton,
  secondaryButton,
}: ModalProps) => {
  const handleClose = () => {
    handleModalClose(false)
  }
  const classes = useStyles()
  return (
    <div>
      <Dialog
        onClose={handleClose}
        aria-labelledby="customized-dialog-title"
        open={open}
        maxWidth={false}
      >
        <DialogTitle onClose={handleClose} buttonStyle={classes.closeButton}>
          {title}
        </DialogTitle>
        <Divider />
        <DialogContent>{children}</DialogContent>
        <DialogActions>
          {secondaryButton}
          {primaryButton}
        </DialogActions>
      </Dialog>
    </div>
  )
}
Modal.propTypes = {
  handleModalClose: PropTypes.func.isRequired,
  open: PropTypes.bool.isRequired,
  title: PropTypes.string.isRequired,
  children: PropTypes.elementType.isRequired,
  primaryButton: PropTypes.elementType,
  secondaryButton: PropTypes.elementType,
}

Modal.defaultProps = {
  primaryButton: null,
  secondaryButton: null,
}

export default Modal
