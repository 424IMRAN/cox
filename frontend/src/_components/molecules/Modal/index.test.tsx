import React from 'react'
import { mount, shallow } from 'enzyme'
import Dialog from '@material-ui/core/Dialog'
import Modal from '.'
import TextField from '../../atoms/TextField'

describe('Text Field  Component', () => {
  it('Renders correctly', () => {
    const wrapper = shallow(
      <Modal handleModalClose={() => {}} open title="test">
        <TextField label="Filter Name" fullWidth />
      </Modal>
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('Allows us to set props', () => {
    const wrapper = mount(
      <Modal handleModalClose={() => {}} open title="test">
        <TextField label="Filter Name" fullWidth />
      </Modal>
    )
    expect(wrapper.props().title).toEqual('test')
    wrapper.setProps({ title: 'foo' })
    expect(wrapper.find(TextField).props().label).toEqual('Filter Name')
    expect(wrapper.props().title).toEqual('foo')
  })

  it('onChange works', () => {
    const mockHandleChangeText = jest.fn()
    const component = mount(
      <Modal handleModalClose={mockHandleChangeText} open title="test">
        <TextField label="Filter Name" fullWidth />
      </Modal>
    )
    component
      ?.find(Dialog)
      ?.props()
      ?.onClose?.(
        {
          target: { checked: true },
        } as React.ChangeEvent<HTMLInputElement>,
        'backdropClick'
      )
    expect(mockHandleChangeText).toHaveBeenCalledWith(false)
  })
})
