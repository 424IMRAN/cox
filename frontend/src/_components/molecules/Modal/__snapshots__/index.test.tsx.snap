// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Field  Component Renders correctly 1`] = `
<div>
  <WithStyles(ForwardRef(Dialog))
    aria-labelledby="customized-dialog-title"
    maxWidth={false}
    onClose={[Function]}
    open={true}
  >
    <DialogTitle
      buttonStyle="makeStyles-closeButton-2"
      onClose={[Function]}
    >
      test
    </DialogTitle>
    <WithStyles(ForwardRef(Divider)) />
    <WithStyles(ForwardRef(DialogContent))>
      <TextField
        customStyle={null}
        endAdorment={null}
        fullWidth={true}
        handleChange={null}
        helperText=""
        label="Filter Name"
        margin="none"
        multiline={false}
        name=""
        placeholder=""
        rows={1}
        rowsMax={1}
        value=""
        variant="outlined"
      />
    </WithStyles(ForwardRef(DialogContent))>
    <WithStyles(ForwardRef(DialogActions)) />
  </WithStyles(ForwardRef(Dialog))>
</div>
`;
